### PR TITLE
Created Storer interface

### PR DESCRIPTION
**Pull request by braddotcoffee**
Storer interface will be used to support a variety of storage
back-ends, such as writing to files or a databse.

### CODE REVIEW

File: storer/storer.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package storer
+
+import "encoding/json"
+
+// Storer writes data collected from Collector to some form
+// of persistent storage
+type Storer interface {
+	saveEntry(json json.Encoder, tag string) error
```
Comment (braddotcoffee) [inline storer/storer.go]: I figure that the API user providing a unique tag to save the data under is the easiest way to make sure that things are saved properly. For MongoDB that'd be used as a table name, for files it'd be the directory to create. The CLI can just have a `-tag`

File: storer/storer.go [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package storer
+
+import "encoding/json"
+
+// Storer writes data collected from Collector to some form
+// of persistent storage
+type Storer interface {
+	initialize() error
```
Comment (braddotcoffee) [inline storer/storer.go:8]: Optional initialization for setting up database connection, etc.

File: storer/storer.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+	saveEntry(json json.Encoder, tag string) error
+	saveEntries(json json.Encoder, tag string) error
```
Comment (hsadoyan) [inline storer/storer.go]: If both are encoding json, do we really need both plural and singular?

**Review by hsadoyan — COMMENTED** [review]
Gitignore the binary too

Wait nvm

File: storer/storer.go (hunk 3 above)
Comment (braddotcoffee) [reply storer/storer.go]: In my mind, yes. Reason being that for files, you will save each of those to a separate file whereas for Mongo it probably matters less

Comment (braddotcoffee) [reply storer/storer.go]: I guess you're right that I don't necessarily see a case where you're really only saving one entry. I can remove the singular and we can just make helper functions if we think we need to

Comment (hsadoyan) [reply storer/storer.go]: Yeah I like that. Have the singular be a helper. And potentially change the plural to take an array of json, cause then we get the type checking. This way if someone calls `saveEntries` with a singular Val we'll have to handle the case manually in the function. Or it becomes a runtime error.

**Comment by hsadoyan** [comment]
🛳

**Pull request merged**